src/Stratis.Features.Collateral/CheckCollateralFullValidationRule.cs

```diff
@@ -48,24 +54,64 @@ public override Task RunAsync(RuleContext context)
                 return Task.CompletedTask;
             }
 
-            // If the time on this block is earlier than the STRAX genesis block then ignore this rule.
-            uint counterChainGenesisBlockTime = this.collateralChecker.GetCounterChainGenesisBlockTime();
-            if (context.ValidationContext.ChainedHeaderToValidate.Header.Time < counterChainGenesisBlockTime)
-                return Task.CompletedTask;                
-
             var commitmentHeightEncoder = new CollateralHeightCommitmentEncoder(this.Logger);
-            (int? commitmentHeight, _) = commitmentHeightEncoder.DecodeCommitmentHeight(context.ValidationContext.BlockToValidate.Transactions.First());
+            (int? commitmentHeight, uint? magic) = commitmentHeightEncoder.DecodeCommitmentHeight(context.ValidationContext.BlockToValidate.Transactions.First());
             if (commitmentHeight == null)
             {
                 // We return here as it is CheckCollateralCommitmentHeightRule's responsibility to perform this check.
                 this.Logger.LogTrace("(-)SKIPPED_AS_COLLATERAL_COMMITMENT_HEIGHT_MISSING]");
                 return Task.CompletedTask;
             }
 
-            this.Logger.LogDebug("Commitment is: {0}.", commitmentHeight);
+            this.Logger.LogDebug("Commitment is: {0}. Magic is: {1}", commitmentHeight, magic);
 
-            // TODO: Both this and CollateralPoAMiner are using this chain's MaxReorg instead of the Counter chain's MaxReorg. Beware: fixing requires fork.
+            // TODO: The code contained in the following "if" can be removed after most nodes have switched their mainchain to Strax.
 
+            // Strategy:
+            // 1. I'm a Cirrus miner on STRAX. If the block's miner is also on STRAX then check the collateral. Pass or Fail as appropriate.
+            // 2. The block miner is on STRAT. If most nodes were on STRAT(prev round) then they will check the rule. Pass the rule.
+            // 3. The miner is on STRAT and most nodes were on STRAX(prev round).Fail the rule.
+
+            // 1. If the block miner is on STRAX then skip this code and go check the collateral.
+            Network counterChainNetwork = this.fullNode.NodeService<CounterChainNetworkWrapper>().CounterChainNetwork;
+            if (this.network.Name.StartsWith("Cirrus") && magic != counterChainNetwork.Magic)
+            {
+                // 2. The block miner is on STRAT.
+                IConsensusManager consensusManager = this.fullNode.NodeService<IConsensusManager>();
+                int memberCount = 0;
+                int membersOnStratis = 0;
+                uint targetSpacing = this.slotsManager.GetRoundLengthSeconds();
+                ChainedHeader chainedHeader = context.ValidationContext.ChainedHeaderToValidate;
+                do
+                {
+                    chainedHeader = chainedHeader.Previous;
```

Comment:
Why are we starting on `Previous`? Should this not be set after the `memberCount` increment?

Reply:
The idea was not to include the block being validated into the prior round being sampled. Changed.